### PR TITLE
many: refactor tpm seal parameter setting

### PR DESCRIPTION
**Pull request by cmatsuoka**
We originally set TPM seal parameters such as keyfile pathnames in
the devicestate install handler to pass it to "snap-bootstrap
create-partitions", which is no longer necessary. With sealing
code moving from the installer to boot, it's cleaner to set them
one level below.
    
Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

### CODE REVIEW

File: gadget/install/install.go [hunk 1]
```diff
@@ -94,11 +90,11 @@ func Run(gadgetRoot, device string, options Options, observer gadget.ContentObse
 	// at this point we removed any existing partition, nuke any
 	// of the existing sealed key files placed outside of the
 	// encrypted partitions (LP: #1879338)
-	if options.KeyFile != "" {
-		if err := os.Remove(options.KeyFile); err != nil && !os.IsNotExist(err) {
-			return fmt.Errorf("cannot cleanup obsolete key file: %v", options.KeyFile)
+	sealedKeyFiles, _ := filepath.Glob(filepath.Join(boot.InitramfsEncryptionKeyDir, "*.sealed-key"))
+	for _, keyFile := range sealedKeyFiles {
```
Comment (bboozzoo) [inline gadget/install/install.go:94]: BTW is there anything else we want to preserve in that directory if we are reinstalling/installing from scratch?

Comment (cmatsuoka) [reply gadget/install/install.go:94]: With the current directory layout no, but if we ever set `InitramfsEncryptionKeyDir` to a different location we may need to keep some files in place.

File: gadget/install/options.go [hunk 2]
```diff
@@ -29,14 +29,6 @@ type Options struct {
...
 	Encrypt bool
 	// KeyFile is the location where the encryption key is written to
-	KeyFile string
-	// RecoveryKeyFile is the location where the recovery key is written to
-	RecoveryKeyFile string
-	// TPMLockoutAuthFile is the location where the TPM lockout authorization is written to
-	TPMLockoutAuthFile string
-	// TPMPolicyUpdateDataFile is the location where the TPM authorization policy update data is written to
-	TPMPolicyUpdateDataFile string
-	// KernelPath is the path to the kernel to seal the keyfile to
```
[the remark this answers is not in the document]
Comment (cmatsuoka) [reply gadget/install/options.go:39]: Fixed, thanks.

**Review by mvo5 — APPROVED** [review]
LGTM

**Pull request merged**